src/charm.py

```diff
@@ -430,6 +434,24 @@ def _bind_address(self) -> Optional[str]:
         except ModelError:
             return None
 
+    @property
+    def _ingress_address(self) -> Optional[str]:
```

Comment:
We will soon add integration with an ingress (traefik) through the [ingress](https://github.com/canonical/charm-relation-interfaces/tree/main/interfaces/ingress/v2) charm relation interface. Will we have 2 meanings for "ingress" or is it the same meaning but this `_ingress_address` returns something else when it's related to an ingress?

Reply:
Indeed, it can be confusing.

In theory, we have one meaning of ingress:

The case of wanting an unique URL to contact the Vault service (not a Vault unit).

Which is served by either the Juju's concept of ingress address or Traefik's ingress.


I believe it's fine to have the ingress property return either the traefik's ingress address when related to an ingress or return juju's concept of ingress address when it's missing.

:warning: 
This just means that in case of CMR that's cross cloud (Vault hosted on a k8s and requiring on any other cloud including an other k8s), the relation cannot work without traefik's ingress. Which is fine if we can alert that we're in a CMR (which I'm not sure is possible) or it's explicitly documented.

Reply:
Not sure how to handle it from a TLS perspective. It depends if traefik is configured in passthrough or reencrypt, but this should be determined when integrating with traefik.